test/fail_compilation/warn12809.d

```diff
@@ -32,3 +32,42 @@ void test_unrachable3()
 
     int x = 1;      // unreachable
 }
+
+/********************************************/
+
+/*
+TEST_OUTPUT:
+---
+fail_compilation/warn12809.d(108): Warning: statement is not reachable
+fail_compilation/warn12809.d(115): Warning: statement is not reachable
+fail_compilation/warn12809.d(122): Warning: statement is not reachable
```

Comment:
The warning message is too vague. How about "Warning: statement *insert statement* is not reachable"?

Reply:
We've had that same message for many years. A better message would be off topic for this PR, but you can submit one that addresses that if you like.